### PR TITLE
JIRA 7+ support

### PR DESCRIPTION
**Pull request by mvdkleijn**
JIRA 7.x support, including flavors ('software' & 'core')
Fix tests for above support

### CODE REVIEW

File: libraries/jira.rb [hunk 1]
```diff
@@ -188,6 +213,20 @@ def jira_checksum_map
           'x32' => 'dc807ebed5065416eebb117c061aa57bd07c1d168136aca786ae2b0c100f7e30',
           'x64' => '9897ae190a87a61624d5a307c428e8f4c86ac9ff03e1a89dbfb2da5f6d3b0dbd',
           'tar' => 'a77cf4c646d3f49d3823a5739daea0827adad1254dae1d1677c629e512a7afd4'
+        },
+        'core' => {
```
Comment (legal90) [inline libraries/jira.rb]: I recommend here either to use new hashes for JIRA 7+ checksums, or just to flap the nesting:

```ruby
'6.4.12' => {
  'x32' => 'dc807ebed5065416eebb117c061aa57bd07c1d168136aca786ae2b0c100f7e30',
  'x64' => '9897ae190a87a61624d5a307c428e8f4c86ac9ff03e1a89dbfb2da5f6d3b0dbd',
  'tar' => 'a77cf4c646d3f49d3823a5739daea0827adad1254dae1d1677c629e512a7afd4'
},
'7.0.0' => {
  'core' => {
    'x32' => 'bcd4746dcd574532061f79ec549e16d8641346f4e45f1cd3db032730fd23ea80',
    'x64' => '314bb496b7d20fb1101eb303c48a80041775e4fadd692fd97583b9c248df5099',
    'tar' => '56bdae7b78ac4472e6c9a22053e4b083d9feb07ee948f4e38c795591d9fc9ae9'
   },
  'software' => {
    'x32' => '3a43274bc2ae404ea8d8c2b50dcb00cc843d03140c5eb11de558b3025202a791',
    'x64' => '49e12b2ba9f1eaa4ed18e0a00277ea7be19ffd6c55d4a692da3e848310815421',
    'tar' => '2eb0aff3e71272dc0fd3d9d6894f219f92033d004e46b25b542241151a732817'
  }
}
```

File: libraries/jira.rb [hunk 2]
```diff
@@ -58,20 +58,39 @@ def jira_version
...
+      version  = node['jira']['version']
+
+      # JIRA versions >= 7.0.0 have different flavors
+      # Also (at this time) the URLs for flavors unfortunately differ
+      if Gem::Version.new(version) < Gem::Version.new(7)
+        product = "#{base_url}/atlassian-jira-#{version}"
+      else
+        case node['jira']['flavor']
+        when 'software'
+          product = "#{base_url}/atlassian-jira-#{node['jira']['flavor']}-#{node['jira']['version']}-jira-#{version}"
```
Comment (legal90) [inline libraries/jira.rb]: JFYI: There are both `node['jira']['version']` and `version` names in the single line. May be, it's better just to use `version` ?

**Comment by legal90** [comment]
@mvdkleijn Cool! :tada:

File: templates/default/tomcat/server.xml.erb [hunk 3]
```diff
@@ -29,10 +29,9 @@
 -->
 <Server port="8005" shutdown="SHUTDOWN">
 
+    <%= render "tomcat/server.jira-#{@template_partial_version}.fragment.xml.erb", option: {} %>
```
Comment (legal90) [inline templates/default/tomcat/server.xml.erb]: Is not it easer just to add "if" statement here?

Comment (patcon) [inline templates/default/tomcat/server.xml.erb]: Yeah, that was my doing from another PR, but I can understand a preference either way. Seemed the proper place to use template partials at the time, but perhaps it's simpler to keep it all in one conditional-filled template :)

Comment (mvdkleijn) [inline templates/default/tomcat/server.xml.erb]: I was not sure myself about this one. I decided to go with the templates but ideally I'd not like to use any template. I know templates are generally the recommended way to do things in Chef, but I don't like overruling files that come with the app such as in this case.

My own preference goes to KISS and initially I thought two separate templates would be more KISS than conditionals, but perhaps I was wrong. I'll change it.

File: libraries/jira.rb (hunk 1 above)
Comment (mvdkleijn) [inline libraries/jira.rb]: Flipping the nesting.

File: libraries/jira.rb (hunk 2 above)
Comment (mvdkleijn) [inline libraries/jira.rb]: Good catch. Left over from rewrite.

**Pull request merged**